src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -853,18 +855,28 @@ public synchronized InputStream getInputStream(ZipEntry ze)
 
         // wrap a verifier stream around the real stream
         return new JarVerifier.VerifierStream(
-            getManifestFromReference(),
-            verifiableEntry(ze),
-            super.getInputStream(ze),
-            jv);
+                getManifestFromReference(),
+                verifiableEntry(ze),
+                super.getInputStream(ze),
+                jv);
+
     }
 
-    private JarEntry verifiableEntry(ZipEntry ze) {
+    private JarEntry verifiableEntry(ZipEntry ze) throws ZipException {
         if (ze instanceof JarFileEntry) {
             // assure the name and entry match for verification
             return ((JarFileEntry)ze).realEntry();
         }
-        ze = getJarEntry(ze.getName());
+        // ZipEntry::getName should not return null
+        if(ze.getName() != null) {
```

Comment:
Nit, add space after "if"

Reply:
if ZipEntry is extended and getName() overridden then you can't trust the name. So I think you'll have extract the name rather than calling ZipEntry::getName twice. I'm almost tempted to have getInputStream(ZipEntry) be re-specified to throw IAE if the zip entry is null.

Reply:
Will do.

Reply:
> I'm almost tempted to have getInputStream(ZipEntry) be re-specified to throw IAE if the zip entry name is null.

I personally think it is best to continue throw the NPE as that provides symmetry with ZipFile::getInputStream,  aligns with the current javadoc where a null parameter will throw an NPE unless specified elsewhere,  there are existing tests which check for an NPE if JarFile::getInpuStream(null) is called.

Reply:
Ah, OK, sorry I was focused on the NPE and uber focused on the `ZipEntry` passed to `getInputStream`.

If you *prefer* an IAE, I can change that when `ZipEntry::getName` returns null.  I do think we are technically covered via `ZipException` which is defined : ` if a zip file format error has occurred` and unless something has gone amiss with the Zip/Jar LOC/CEN, you should expect a non-null value.

Please let me know your preference as if we go the IAE route, I will need to create a CSR which I was hoping to avoid